javaee-cafe/src/main/java/cafe/web/rest/CafeResource.java

```diff
@@ -18,51 +18,65 @@
 public class CafeResource {
 
-	private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
 
-	@Inject
-	private CafeRepository cafeRepository;
+    @Inject
+    private CafeRepository cafeRepository;
+    
+    @Inject
+    @ConfigProperty(name = "admin.group.id")
+    private String ADMIN_GROUP_ID;
 
-	@GET
-	@Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
-	public List<Coffee> getAllCoffees() {
-		return this.cafeRepository.getAllCoffees();
-	}
+    @Inject
+    private JsonWebToken jwtPrincipal;
+    
+    @GET
+    @Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
+    public List<Coffee> getAllCoffees() {
+        return this.cafeRepository.getAllCoffees();
+    }
 
-	@POST
-	@Consumes({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
-	@Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })	
-	public Coffee createCoffee(Coffee coffee) {
-		try {
-			return this.cafeRepository.persistCoffee(coffee);
-		} catch (PersistenceException e) {
-			logger.log(Level.SEVERE, "Error creating coffee {0}: {1}.", new Object[] { coffee, e });
-			throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
-		}
-	}
+    @POST
+    @Consumes({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
+    public Coffee createCoffee(Coffee coffee) {
+        try {
+            return this.cafeRepository.persistCoffee(coffee);
+        } catch (PersistenceException e) {
+            logger.log(Level.SEVERE, "Error creating coffee {0}: {1}.", new Object[] { coffee, e });
+            throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-	@GET
-	@Path("{id}")
-	@Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
-	public Coffee getCoffeeById(@PathParam("id") Long coffeeId) {
-		return this.cafeRepository.findCoffeeById(coffeeId);
-	}
+    @GET
+    @Path("{id}")
+    @Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON })
+    public Coffee getCoffeeById(@PathParam("id") Long coffeeId) {
+        return this.cafeRepository.findCoffeeById(coffeeId);
+    }
 
-	@DELETE
-	@Path("{id}")
-	public void deleteCoffee(@PathParam("id") Long coffeeId) {
-		try {
-			this.cafeRepository.removeCoffeeById(coffeeId);
-		} catch (IllegalArgumentException ex) {
-			logger.log(Level.SEVERE, "Error calling deleteCoffee() for coffeeId {0}: {1}.",
-					new Object[] { coffeeId, ex });
-			throw new WebApplicationException(Response.Status.NOT_FOUND);
-		}
-	}
+    @DELETE
+    @Path("{id}")
+    public void deleteCoffee(@PathParam("id") Long coffeeId) {
+        if (!this.jwtPrincipal.getGroups().contains(ADMIN_GROUP_ID)) {
+            throw new WebApplicationException(Response.Status.FORBIDDEN);
+        }
```

Comment:
RBAC using `groups` claim from JWT.
BTW, this is the major change of this file, others are formatting changes.